plugins/async-node/core/src/index.test.ts

```diff
@@ -1,42 +1,134 @@
-  id: "test-flow",
+  id: 'test-flow',
   views: [
     {
-      id: "my-view",
+      id: 'my-view',
       actions: [
         {
           asset: {
-            id: "action-0",
-            type: "action",
-            value: "{{foo.bar}}",
-          },
+            id: 'action-0',
+            type: 'action',
+            value: '{{foo.bar}}'
+          }
         },
         {
-          id: "uhh",
-          async: "true",
-        },
-      ],
-    },
+          id: 'uhh',
+          async: 'true'
+        }
+      ]
+    }
   ],
   navigation: {
-    BEGIN: "FLOW_1",
+    BEGIN: 'FLOW_1',
     FLOW_1: {
-      startState: "VIEW_1",
+      startState: 'VIEW_1',
       VIEW_1: {
-        state_type: "VIEW",
-        ref: "my-view",
-        transitions: {},
-      },
-    },
-  },
+        state_type: 'VIEW',
+        ref: 'my-view',
+        transitions: {}
+      }
+    }
+  }
+};
+
+const asyncNodeTest = async (resolvedValue: any, expectedActionType: string) => {
+  const plugin = new AsyncNodePlugin({
+    plugins: [new AsyncNodePluginPlugin()]
+  });
+
+  let deferredResolve: ((value: any) => void) | undefined;
+
+  let resolverInstance: Resolver;
+
+  plugin.hooks.onAsyncNode.tap('test', async (node: Node.Node) => {
+    return new Promise((resolve) => {
+      deferredResolve = resolve; // Promise would be resolved only once
+    });
+  });
+
+  let updateNumber = 0;
+
+  const player = new Player({ plugins: [plugin] });
+
+  player.hooks.viewController.tap('async-node-test', (vc) => {
+    vc.hooks.view.tap('async-node-test', (view) => {
+      view.hooks.onUpdate.tap('async-node-test', () => {
+        updateNumber++;
+      });
+    });
+  });
+
+  player.hooks.viewController.tap("async-node-test", (vc) => {
+    vc.hooks.view.tap("async-node-test", (view) => {
+      view.hooks.resolver.tap("async-node-test", (resolver) => {
+        resolverInstance = resolver;
+      });
+    });
+  });
+
+  player.start(basicFRFWithActions as any);
+
+  let view = (player.getState() as InProgressState).controllers.view.currentView
+    ?.lastUpdate;
+
+  expect(view).toBeDefined();
+  expect(view?.actions[0].asset.type).toBe('action');
+  expect(view?.actions[1]).toBeUndefined();
+
+  await waitFor(() => {
+    expect(deferredResolve).toBeDefined();
+  });
+
+  // Consumer responds with null/undefined
+  if (deferredResolve) {
+    deferredResolve(resolvedValue);
+  }
+
+  await waitFor(() => {
+    expect(updateNumber).toBe(2);
+  });
+
+  view = (player.getState() as InProgressState).controllers.view.currentView
+    ?.lastUpdate;
+
+  expect(view?.actions[0].asset.type).toBe('action');
+  expect(view?.actions[1]?.asset.type).toBe(expectedActionType);
+  expect(updateNumber).toBe(2); // Replace with the actual expected number of updates
+
+  await waitFor(() => {
+    expect(resolverInstance).toBeDefined();
+  });
+
+  // Now we will force update the view using the resolver instance
+  resolverInstance.update();
```

Comment:
this is where you would update using the view instead of the resolver, since if you search for `onUpdate.call` in the code, you'll only find it in the view update scope, not the resolver update scope

Reply:
done